include/glow/Support/ThreadPool.h

```diff
@@ -38,7 +39,20 @@ class ThreadPool final {
   ~ThreadPool();
 
   /// Submit \p fn as a work item for the thread pool.
-  void submit(const std::function<void(void)> &fn);
+  /// \p fn must be a lambda with void return type and arguments.
+  template <typename F> std::future<void> submit(F &&fn) {
```

Comment:
Did you mean to put the implementation of this in the header? Can you move it to the source file?

Reply:
It's a templated method, we can't implement it in the source file without binding the template parameter, which for lambdas is inconvenient.